packages/fx-core/src/common/spec-parser/specParser.ts

```diff
@@ -171,8 +172,13 @@ export class SpecParser {
     await this.loadSpec();
     const newUnResolvedSpec = await specFilter(filter, this.unResolveSpec!);
     if (outputSpecPath) {
-      const ymlString = jsyaml.dump(newUnResolvedSpec);
-      await fs.writeFile(outputSpecPath, ymlString);
+      let resultStr;
+      if (await isYamlSpecFile(this.specPath)) {
```

Comment:
I was expecting this method was exposed so that we could determine the outputSpecPath (whether end with .json or .yml)

Reply:
Sure.
As discussed, yuqi will use this function to determine the outputSpecPath's extension name, and pass it into generate function, so inside the generate function, check the extension name is enough.